src/java/org/apache/cassandra/service/StartupChecks.java

```diff
@@ -339,8 +350,49 @@ public void execute()
             long maxMapCount = getMaxMapCount();
             if (maxMapCount < EXPECTED_MAX_MAP_COUNT)
                 logger.warn("Maximum number of memory map areas per process (vm.max_map_count) {} " +
-                            "is too low, recommended value: {}, you can change it with sysctl.",
-                            maxMapCount, EXPECTED_MAX_MAP_COUNT);
+                            "is too low, recommended value: {}, you can change it with sysctl. {}",
+                            maxMapCount, EXPECTED_MAX_MAP_COUNT, WARN_SUFFIX);
+        }
+    };
+
+    public static final StartupCheck checkYamlConfig = new StartupCheck()
+    {
+        public void execute() throws StartupException
+        {
+            if (Murmur3Partitioner.instance != DatabaseDescriptor.getPartitioner())
+                logger.warn("Not using murmur3 partitioner ({}). {}", DatabaseDescriptor.getPartitioner(), WARN_SUFFIX);
+
+            if (DatabaseDescriptor.getNumTokens() > 16)
+                logger.warn("num_tokens {} too high. values over 16 poorly impact repairs and node bootstrapping/decommissioning. {}",
+                            DatabaseDescriptor.getNumTokens(), WARN_SUFFIX);
+
+            // ServerTestUtils.prepare() enables transient replication, so we have to skip this when we're inside a unit test
+            if(DatabaseDescriptor.isTransientReplicationEnabled() && null == System.getProperty("cassandra.testtag"))
+                throw new StartupException(StartupException.ERR_WRONG_CONFIG, "Transient Replication cannot not be used in HCD.");
+
+            if(DatabaseDescriptor.getEnableMaterializedViews())
+                logger.warn("Materialised Views should not be used in production. {}", WARN_SUFFIX);
+
+            if(DatabaseDescriptor.getEnableSASIIndexes())
+                logger.warn("SASI should not be used in production. {}", WARN_SUFFIX);
+
+            if(DatabaseDescriptor.enableDropCompactStorage())
+                logger.warn("Dropping COMPACT storage tables should not be used in production. {}", WARN_SUFFIX);
```

Comment:
Was this more elaborate? Like CS support is extremely shallow, not recommended and yes dropping them was a problem whose specifics I can't remember. But I think CS support should come with an extra 4 o 5 warnings. Should we ping Ekaterina or are you intentionally only targetting CS drops here?

Reply:
That would be a separate different check.  That would need to scan through all the tables and check if it's COMPACT.  

I can do that (in `checkTableCompactionStrategies()`, changing its method name to `checkTableSetting()`, BUT there the message would have to be "Do not use COMPACT, and do not `drop compact`, instead manually migrate the table to a new non-compact table"

Reply:
I'd say we want to do that now you have the code handy unless there's any heavy-lifting involved I am missing to spot? Or if you don't agree or don't see value?

Reply:
done